ballerina/tests/55_declarative_authentication_service.bal

```diff
@@ -0,0 +1,161 @@
+                password: "ballerina"
+            },
+            certAlias: "ballerina"
+        }
+    },
+    scopeKey: "scope"
+};
+
+OAuth2IntrospectionConfig oauth2IntroConfig55 = {
+    url: "https://localhost:" + oauth2AuthorizationServerPort.toString() + "/oauth2/token/introspect",
+    tokenTypeHint: "access_token",
+    scopeKey: "scp",
+    clientConfig: {
+        secureSocket: {
+            cert: {
+                path: TRUSTSTORE_PATH,
+                password: "ballerina"
+            }
+        }
+    }
+};
+
+LdapUserStoreConfig ldapUserStoreConfig = {
+    domainName: "avix.lk",
+    connectionUrl: "ldap://localhost:389",
+    connectionName: "cn=admin,dc=avix,dc=lk",
+    connectionPassword: "avix123",
+    userSearchBase: "ou=Users,dc=avix,dc=lk",
+    userEntryObjectClass: "inetOrgPerson",
+    userNameAttribute: "uid",
+    userNameSearchFilter: "(&(objectClass=inetOrgPerson)(uid=?))",
+    userNameListFilter: "(objectClass=inetOrgPerson)",
+    groupSearchBase: ["ou=Groups,dc=avix,dc=lk"],
+    groupEntryObjectClass: "groupOfNames",
+    groupNameAttribute: "cn",
+    groupNameSearchFilter: "(&(objectClass=groupOfNames)(cn=?))",
+    groupNameListFilter: "(objectClass=groupOfNames)",
+    membershipAttribute: "member",
+    userRolesCacheEnabled: true,
+    connectionPoolingEnabled: false,
+    connectionTimeout: 5,
+    readTimeout: 60
+};
+
+JwtValidatorConfigWithScopes jwtAuthConfig55WithScopes = {
+    jwtValidatorConfig: jwtAuthConfig55,
+    scopes: "write"
+};
+
+JwtValidatorConfigWithScopes jwtAuthConfig55EmptyScope = {
+    jwtValidatorConfig: jwtAuthConfig55
+};
+
+OAuth2IntrospectionConfigWithScopes oauth2config55WithScopes = {
+    oauth2IntrospectionConfig: oauth2IntroConfig55,
+    scopes: "write"
+};
+
+OAuth2IntrospectionConfigWithScopes oauth2config55EmptyScope = {
+    oauth2IntrospectionConfig: oauth2IntroConfig55
+};
+
+LdapUserStoreConfigWithScopes ldapUserStoreconfig55WithScopes = {
+    ldapUserStoreConfig: ldapUserStoreConfig,
+    scopes: "developer"
+};
+
+LdapUserStoreConfigWithScopes ldapUserStoreconfig55EmptyScope = {
+    ldapUserStoreConfig: ldapUserStoreConfig
+};
+
+FileUserStoreConfigWithScopes fileUserStoreConfig55WithScopes = {
+    fileUserStoreConfig: {},
+    scopes: "write"
+};
+
+FileUserStoreConfigWithScopes fileUserStoreConfig55EmptyScope  = {
+    fileUserStoreConfig: {}
+};
+
+@ServiceConfig {
+    auth: [
+        jwtAuthConfig55WithScopes,
+        oauth2config55WithScopes,
+        // ldapUserStoreconfig55WithScopes,
+        fileUserStoreConfig55WithScopes
+    ]
+}
+@ServiceDescriptor {descriptor: ROOT_DESCRIPTOR_55, descMap: getDescriptorMap55()}
+service "helloWorld55" on ep55WithScopes {
+
+    remote function hello55BiDiWithCaller(HelloWorld55StringCaller caller,
+     stream<string, Error?> clientStream) returns error? {
+        authenticateResource(self); // Currently the desugar function is called manually until the compiler supports it
```

Comment:
Once we remove these lines by updating to relevant language version, this PR looks okay.

Reply:
Removed in https://github.com/ballerina-platform/module-ballerina-grpc/pull/343/commits/014eddd5bf8cc917667ab4792cf5666e871718eb